ckks/keygen.go

```diff
@@ -28,248 +24,16 @@ type KeyGenerator interface {
 // KeyGenerator is a structure that stores the elements required to create new keys,
 // as well as a small memory pool for intermediate values.
 type keyGenerator struct {
```

Comment:
Let's discuss if needed. We could have a `GenRelinearizationKeyForCiphertextDegree` method in the `rlwe.KeyGenerator` interface and it would re-align the two schemes.

Reply:
Modification in the evaluator would need to be done to accommodate for arbitrary degree tensoring and relinearization